check-github-user-permission/index.ts

```diff
@@ -0,0 +1,54 @@
+import * as core from "@actions/core";
+import { getOctokit } from "@actions/github";
+
+async function isUserInGithubTeam(token: string, orgName: string, teamSlug: string, username: string): Promise<boolean> {
+    const octokit = getOctokit(token);
+
+    try {
+        const teamResponse = await octokit.request('GET /orgs/{org}/teams/{team_slug}', {
+            org: orgName,
+            team_slug: teamSlug,
+            headers: {
+                'X-GitHub-Api-Version': '2022-11-28'
+            }
+        })
+        core.debug(`Find team ${teamResponse.data.name} in org ${teamResponse.data.organization}`)
+    } catch (e: any) {
+        if (e.status == 404) {
+            throw new Error(`Team ${teamSlug} not found in ${orgName}`);
+        }
+        throw new Error(`Got ${e.status} ${e.response} to check team ${teamSlug} in ${orgName}`);
+    }
+
+    try {
+        const membershipResponse = await octokit.request('GET /orgs/{org}/teams/{team_slug}/memberships/{username}', {
```

Comment:
suggest to use rest methods from `@octokit` packages like [here](https://github.com/JamesSingleton/is-organization-member/blob/main/src/main.js#L19), to simpify the code.

https://github.com/octokit/plugin-rest-endpoint-methods.js/blob/main/src/generated/endpoints.ts#L1597

Reply:
looks good. let me check